components/tensorboard-controller/controllers/tensorboard_controller.go

```diff
@@ -126,27 +130,69 @@ func (r *TensorboardReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func generateDeployment(tb *tensorboardv1alpha1.Tensorboard, log logr.Logger) *appsv1.Deployment {
+func generateDeployment(tb *tensorboardv1alpha1.Tensorboard, log logr.Logger, r *TensorboardReconciler) (*appsv1.Deployment, error) {
 	var volumeMounts []corev1.VolumeMount
 	var volumes []corev1.Volume
+	var mountpath, subpath string = tb.Spec.LogsPath, ""
+	var affinity = &corev1.Affinity{}
 
-	//In this case, a PVC is used as a log storage for the Tensorboard server
+	//In this case, a PVC is used as a log storage for the Tensorboard server.
 	if !isCloudPath(tb.Spec.LogsPath) {
+		var pvcname string
+
+		//General case, in which tb.Spec.LogsPath follows the format: "pvc://<pvc-name>/<local-path>".
+		if isPVCPath(tb.Spec.LogsPath) {
+			pvcname = extractPVCName(tb.Spec.LogsPath)
+			mountpath = "/tensorboard_logs/"
+			subpath = extractPVCSubPath(tb.Spec.LogsPath)
+		} else {
+			//Maintaining backwards compatibility with previous version of the controller.
+			pvcname = "tb-volume"
+			subpath = ""
+		}
+
 		volumeMounts = append(volumeMounts, corev1.VolumeMount{
 			Name:      "tbpd",
 			ReadOnly:  true,
-			MountPath: tb.Spec.LogsPath,
+			MountPath: mountpath,
+			SubPath:   subpath,
 		})
 		volumes = append(volumes, corev1.Volume{
 			Name: "tbpd",
 			VolumeSource: corev1.VolumeSource{
 				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
-					ClaimName: "tb-volume",
+					ClaimName: pvcname,
 				},
 			},
 		})
+
+		if err, sch := rwoPVCScheduling(); err != nil {
+			return nil, err
+		} else if err == nil && sch {
```

Comment:
nit: I think you don't need the `err == nil` check here since this holds from the previous if check